src/data/media-service.ts

```diff
@@ -0,0 +1,86 @@
+import { decodeMediaData } from '~/utils/decode-media-data'
+import { VersionedApiService } from '~/data/api-service'
+
+import type { ApiQueryParams, MediaResult } from '~/store/types'
+import type { Media } from '~/models/media'
+
+import type { AxiosResponse } from 'axios'
+
+class MediaService<T extends Media> {
+  private readonly mediaType: T['frontendMediaType']
+
+  constructor(mediaType: T['frontendMediaType']) {
+    this.mediaType = mediaType
+  }
+
+  /**
+   * Decodes the text data to avoid encoding problems.
+   * Also, converts the results from an array of media objects into an object with
+   * media id as keys.
```

Comment:
```suggestion
   * Decodes the text data to avoid encoding problems.
   * Also, converts the results from an array of media
   * objects into an object with media id as keys.
```

minor, minor nit to clean up the line breaks here